plugins/state_monitor.py

```diff
@@ -0,0 +1,238 @@
+                    {key: str(value) for key, value in data.items()})
+
+        # Update state for next pass through the pipeline
+        self._previous_data = input_data
+
+        # Pass through the input for consumption by any further steps
+        return input_data
+
+    def send_message(self, msg):
+        """
+        Send a message via the given method.
+
+        This is a shepherd method. Given a generic message, we'll send it
+        using the method specified when initializing the class. Before sending, we'll
+        add in which sensor is sending the message.
+
+        Note: Any "sender class" must have a send method.
+
+        Parameters
+        ----------
+        msg : str
+            The message to be sent.
+
+        """
+        msg = self.construct_message(msg)
+
+        if self.method_cls is not None:
+            self.method_cls.send(msg)
+
+    @classmethod
+    def construct_message(cls, msg):
+        """ Construct the message detailing the state notification. """
+        from brokkr.config.unit import UNIT_CONFIG
+        from brokkr.config.metadata import METADATA
+
+        sensor_name = f"{METADATA['name']}{UNIT_CONFIG['number']:02d}"
+
+        header = f"Message from sensor {sensor_name}"
+
+        msg = '\n'.join([header, msg])
+
+        return msg
+
+
+class SlackSender:
+    def __init__(self, key_file, channel, logger=None):
+        """ 
+        Handle sending messages in Slack. 
+        
+        Parameters
+        ----------
+        key_file : str
+            The path to the file that contains the Slack key for the
+            workspace to send the message.
+        channel : str
+            The channel to send the message to. It should be defined in file
+            provided by `key_file`.
+        logger : Python logger
+            If not None, if message send fails, log it here.
+
+        """
+
+        self.logger = logger
+
+        slack_key_file = Path(key_file).expanduser()
+
+        if not slack_key_file.exists():
+            raise FileNotFoundError
+
+        config = ConfigParser()
+        config.read(slack_key_file)
+
+        # This should be constant...
+        BASE_URL = 'https://hooks.slack.com/services/'
+
+        # Define the url, including the channel....
+        slack_url = urllib.parse.urljoin(BASE_URL, config['channel'][channel])
+
+        self.request = urllib.request.Request(url=str(slack_url),
+                                              headers={'Content-type': 'application/json'},
+                                              method='POST')
+
+    def send(self, msg):
+        """
+        Send a Slack message.
+
+        If the message sending fails and there is a logger associated with
+        the class, log it. Otherwise, raise a `RuntimeWarning`.
+        """
+
+        self.request.data = f'{{"text": "{msg}"}}'.encode('utf-8')
+
+        resp = urllib.request.urlopen(self.request)
+
+        if resp.status != 200:
```

Comment:
Would be a good idea to factor this magic number into a constant, e.g. `HTTP_STATUS_OK`.